src/testpool.jl

```diff
@@ -27,19 +27,25 @@ Base.@kwdef mutable struct TestEnginePool
     creater::Function = create_default_engine
 end
 
+function _get_new_id()
+    return Threads.atomic_add!(TEST_ENGINE_POOL.next_id, 1)
+end
+
 function get_free_test_engine_name()::String
     delay = 1
     # One lock guards name acquisition, forming a queue
     # The second lock guards modification of the engine pool
     @lock TEST_SERVER_ACQUISITION_LOCK while true
```

Comment:
with next id being an atomic, is the `TEST_SERVER_ACQUISITION_LOCK` still needed?

Reply:
`next_id` is for engine naming, so they don't interact. It's there to simplify starting up/naming/renaming multiple engines at once.

`TEST_SERVER_ACQUISITION_LOCK` is there to restrict engine claiming to one test a time - which forms a queue once all engines are claimed. The inner lock then gives 'privileged' access to anything else trying to modify the list. A standard test has to claim both locks, restricting access to the inner lock to one test thread at a time. Thus an attempt to replace an engine has to contend for the lock with only one thread that spends most of its time sleeping. Otherwise attempts to modify the list during testing could wait until all the tests had finished.

Disallowing modification of the list while tests are running would also work, of course.